src/test/spec/unified_runner/operation.rs

```diff
@@ -92,6 +95,22 @@ impl<'de> Deserialize<'de> for OperationObject {
     }
 }
 
+#[derive(Debug)]
+pub enum Expectation {
+    Result {
+        value: Option<Bson>,
+        save_as_entity: Option<String>,
+    },
+    Error(ExpectError),
+    Ignore,
+}
+
+fn deserialize_op<'de, 'a, T: 'a + Deserialize<'de> + TestOperation>(
```

Comment:
I factored this out of the match in `Operation::deserialize` below to make it more clear there that it's just the type parameter that varies.